frame/assets/src/lib.rs

```diff
@@ -901,6 +800,191 @@ impl<T: Config> Module<T> {
 	}
 }
 
+impl<T: Config> FungibleAsset<T::AccountId> for Module<T> where
+	T::Balance: MaybeSerializeDeserialize + Debug
+{
+	type Balance = T::Balance;
+	type AssetId = T::AssetId;
+
+	// PUBLIC IMMUTABLES
+
+	/// The total amount of issuance in the system for a specific asset.
+	fn total_issuance(id: Self::AssetId) -> Self::Balance {
+		Self::total_supply(id)
+	}
+
+	fn balance_of(id: Self::AssetId, who: T::AccountId) -> Self::Balance {
+		Self::balance(id, who)
+	}
+
+	// PUBLIC MUTABLES (DANGEROUS)
+
+	fn transfer(
+		id: Self::AssetId,
+		from: T::AccountId,
+		dest: T::AccountId,
+		amount: Self::Balance
+	) -> DispatchResult {
+		if dest == from {
+			return Ok(())
+		}
+
+		let mut from_account = Account::<T>::get(id, &from);
+		from_account.balance = from_account.balance.checked_sub(&amount)
+			.ok_or(Error::<T>::BalanceLow)?;
+
+		Asset::<T>::try_mutate(id, |maybe_details| {
+			let details = maybe_details.as_mut().ok_or(Error::<T>::Unknown)?;
+
+			let mut amount = amount;
+			if from_account.balance < details.min_balance {
+				amount += from_account.balance;
+				from_account.balance = Zero::zero();
+			}
+
+			Account::<T>::try_mutate(id, &dest, |a| -> DispatchResult {
+				let new_balance = a.balance.saturating_add(amount);
+				ensure!(new_balance >= details.min_balance, Error::<T>::BalanceLow);
+				if a.balance.is_zero() {
+					a.is_zombie = Self::new_account(&dest, details)?;
+				}
+				a.balance = new_balance;
+				Ok(())
+			})?;
+
+			match from_account.balance.is_zero() {
+				false => {
+					Self::dezombify(&from, details, &mut from_account.is_zombie);
+					Account::<T>::insert(id, &from, &from_account)
+				}
+				true => {
+					Self::dead_account(&from, details, from_account.is_zombie);
+					Account::<T>::remove(id, &from);
+				}
+			}
+
+			Self::deposit_event(RawEvent::Transferred(id, from, dest, amount));
+
+			Ok(())
+		})
+	}
+}
+
+impl<T: Config> BurnableAsset<T::AccountId> for Module<T> {
+	fn burn(
+		id: Self::AssetId,
+		who: T::AccountId,
+		amount: Self::Balance
+	) -> DispatchResult {
+		Asset::<T>::try_mutate(id, |maybe_details| {
```

Comment:
if it's the pallet which (through type `T`) can be configured to be non-burnable, then why would this implementation (over `T`) not adhere to that?

Reply:
Not sure I understand entirely, are you saying its best to move these implementations into other repos that want this functionality?